tt_metal/impl/program/program.cpp

```diff
@@ -156,7 +168,7 @@ class Program_ {
     std::vector<std::shared_ptr<Buffer>> owned_buffer_pool = {};
 
     // The buffer that holds the kernel/binaries/etc for this program
-    std::shared_ptr<Buffer> kernels_buffer = nullptr;
+    std::unordered_map<std::size_t, std::shared_ptr<Buffer>> kernels_buffer = {};
```

Comment:
`kernels_buffer_`, also no need for `= {}`

Reply:
Done